lib/matplotlib/figure.py

```diff
@@ -1745,7 +1743,7 @@ def subplots_adjust(self, *args, **kwargs):
             if not isinstance(ax, SubplotBase):
                 # Check if sharing a subplots axis
                 if (ax._sharex is not None and
-                    isinstance(ax._sharex, SubplotBase)):
+                        isinstance(ax._sharex, SubplotBase)):
```

Comment:
Just make this `isinstance(ax._sharex, SubplotBase)` (the first check is not needed).